### PR TITLE
add unique access count metric and metric at artifact level

### PR DESCRIPTION
**Pull request by zhenz**

### CODE REVIEW

File: trovi/api/serializers.py [hunk 1]
```diff
@@ -255,6 +258,29 @@ def to_internal_value(self, data: dict[str, JSON]) -> dict[str, JSON]:
...
+        for v in instance.versions.all():
+            origins = v.events.filter(event_type=ArtifactEvent.EventType.LAUNCH).values(
+                "event_origin"
+            )
+            unique_origins.update([o["event_origin"] for o in origins])
+            total_access_count += v.access_count
+        return {
+            "access_count": total_access_count,
+            "unique_access_count": len(unique_origins),
+        }
```
Comment (super-cooper) [inline trovi/api/serializers.py:281]: `Artifact`s already have an `access_count` field which we can use here. I also checked, and I think it's possible to limit the calculation of `unique_access_count` to a single SQL query, similar to how you did it for `ArtifactVersion`

```suggestion
        return {
            "access_count": total_access_count,
            "unique_access_count": ArtifactEvent.objects.filter(
                artifact_version__artifact=instance,
                event_type=ArtifactEvent.EventType.LAUNCH,
            )
            .values("event_origin")
            .distinct()
            .count(),
        }
```

**Review by super-cooper — CHANGES_REQUESTED** [review]
I like how small the footprint of this change is :) I was worried that it was going to be difficult to fit it in. I have a suggestion for optimization, and to expose the new field to the docs generator.

Also, could you write a simple unit test for this? I think it could be almost exactly the same as the one for [`access_count`](https://github.com/ChameleonCloud/trovi/blob/cc3dd3045c7a65c65a3af9261fdcd3c73c5b42f1/trovi/api/tests.py#L973).

**Comment by zhenz** [comment]
> I like how small the footprint of this change is :) I was worried that it was going to be difficult to fit it in. I have a suggestion for optimization, and to expose the new field to the docs generator.
> 
> Also, could you write a simple unit test for this? I think it could be almost exactly the same as the one for [`access_count`](https://github.com/ChameleonCloud/trovi/blob/cc3dd3045c7a65c65a3af9261fdcd3c73c5b42f1/trovi/api/tests.py#L973).

@super-cooper Thanks! and sorry for taking so long to address your comments. Those are great suggestions!

**Review by super-cooper — APPROVED** [review]
Looks great!

**Pull request merged**